go/test/endtoend/vreplication/global_routing_test.go

```diff
@@ -0,0 +1,314 @@
+		},
+		postKsS1: func(t *testing.T) {
+			require.True(t, h.isGlobal(t, []string{"t2", "t4"}, grTestConfig.ksS1))
+			require.True(t, h.isNotGlobal(t, []string{"t1", "t3"}))
+			require.True(t, h.isNotGlobal(t, []string{"t5"}))
+			require.True(t, h.isGlobal(t, []string{"t6"}, grTestConfig.ksS1))
+		},
+	}
+	exp[grTestCase{unshardedHasVSchema: false, markAsGlobal: true}] = &grTestExpectations{
+		postKsU1: func(t *testing.T) {
+			require.True(t, h.isGlobal(t, []string{"t1", "t2", "t3"}, grTestConfig.ksU1))
+		},
+		postKsU2: func(t *testing.T) {
+			require.True(t, h.isGlobal(t, []string{"t1", "t3"}, grTestConfig.ksU1))
+			require.True(t, h.isGlobal(t, []string{"t4", "t5"}, grTestConfig.ksU2))
+			require.True(t, h.isAmbiguous(t, []string{"t2"}))
+		},
+		postKsS1: func(t *testing.T) {
+			require.True(t, h.isGlobal(t, []string{"t2", "t4"}, grTestConfig.ksS1))
+			require.True(t, h.isGlobal(t, []string{"t1", "t3"}, grTestConfig.ksU1))
+			require.True(t, h.isGlobal(t, []string{"t5"}, grTestConfig.ksU2))
+			require.True(t, h.isGlobal(t, []string{"t6"}, grTestConfig.ksS1))
+		},
+	}
+	exp[grTestCase{unshardedHasVSchema: true, markAsGlobal: false}] = &grTestExpectations{
+		postKsU1: func(t *testing.T) {
+			require.True(t, h.isGlobal(t, []string{"t1", "t2", "t3"}, grTestConfig.ksU1))
+		},
+		postKsU2: func(t *testing.T) {
+			require.True(t, h.isGlobal(t, []string{"t1", "t3"}, grTestConfig.ksU1))
+			require.True(t, h.isGlobal(t, []string{"t4", "t5"}, grTestConfig.ksU2))
+			require.True(t, h.isAmbiguous(t, []string{"t2"}))
+		},
+		postKsS1: func(t *testing.T) {
+			require.True(t, h.isAmbiguous(t, []string{"t2", "t4"}))
+			require.True(t, h.isGlobal(t, []string{"t1", "t3"}, grTestConfig.ksU1))
+			require.True(t, h.isGlobal(t, []string{"t5"}, grTestConfig.ksU2))
+		},
+	}
+	exp[grTestCase{unshardedHasVSchema: true, markAsGlobal: true}] =
+		exp[grTestCase{unshardedHasVSchema: true, markAsGlobal: false}]
+	return &exp
+
+}
+
+func (h *grHelpers) getUnshardedVschema(unshardedHasVSchema bool, tables []string) string {
+	if !unshardedHasVSchema {
+		return ""
+	}
+	vschema := `{"tables": {`
+	for i, table := range tables {
+		if i != 0 {
+			vschema += `,`
+		}
+		vschema += fmt.Sprintf(`"%s": {}`, table)
+	}
+	vschema += `}}`
+	return vschema
+}
+
+func (h *grHelpers) rebuildGraphs(t *testing.T, keyspaces []string) {
+	var err error
+	for _, ks := range keyspaces {
+		err = vc.VtctldClient.ExecuteCommand("RebuildKeyspaceGraph", ks)
+		require.NoError(t, err)
+	}
+	require.NoError(t, err)
+	err = vc.VtctldClient.ExecuteCommand("RebuildVSchemaGraph")
+	require.NoError(t, err)
+
+}
+
+func TestGlobalRouting(t *testing.T) {
+	h := grHelpers{t}
+	exp := *h.getExpectations()
+	testCases := []grTestCase{
+		{unshardedHasVSchema: false, markAsGlobal: true},
+		{unshardedHasVSchema: false, markAsGlobal: false},
+		{unshardedHasVSchema: true, markAsGlobal: true},
+		{unshardedHasVSchema: true, markAsGlobal: false},
+	}
+	for _, tc := range testCases {
```

Comment:
This feels duplicated. The `exp` already has this information in its keys. 
We could just do this instead - 
```go
for tc, funcs := range exp {
		require.NotNil(t, funcs)
		testGlobalRouting(t, tc.markAsGlobal, tc.unshardedHasVSchema, funcs)
	}
```
This avoids the need for explicitly writing testcases.